app/src/main/java/com/lagradost/quicknovel/providers/EfremnetProvider.kt

```diff
@@ -0,0 +1,1502 @@
+        for (h in headers) {
+            val name = h?.selectFirst("h2")?.text() ?: continue
+            val cUrl = h.selectFirst(" div > h2.title > a ")?.attr("href") ?: continue
+
+            val posterUrl = h.selectFirst("div > a.highslide > img")?.attr("src")
+
+            returnValue.add(
+                SearchResponse(
+                    name,
+                    cUrl,
+                    fixUrlNull(posterUrl),
+                    null,
+                    null,
+                    this.name
+                )
+            )
+        }
+        return HeadMainPageResponse(url, returnValue)
+    }
+
+    override suspend fun loadHtml(url: String): String? {
+        val response = app.get(url, headers = baseHeaders)
+        val document = Jsoup.parse(response.text)
+        document.select("div.splitnewsnavigation").remove()
+        document.select("div.splitnewsnavigation2").remove()
+        return document.selectFirst("#textToRead")?.html()
+    }
+
+
+    override suspend fun search(query: String): List<SearchResponse> {
+        val response =
+            app.get(
+                "$mainUrl/build_in_search/?q=$query",
+                headers = baseHeaders
+            ) // AJAX, MIGHT ADD QUICK SEARCH
+
+        val document = Jsoup.parse(response.text)
+
+        val headers = document.select("div > article > div.box_in[id='search result']")
+
+        return headers.mapNotNull { h ->
+            val name = h?.selectFirst(" div > h2.title > a > b")?.text() ?: return@mapNotNull null
+            val cUrl = mainUrl + h.selectFirst(" div > h2.title > a ")?.attr("href")
+
+            val posterUrl = h.selectFirst("div > a.highslide > img")?.attr("src")
+
+            SearchResponse(
+                name,
+                cUrl,
+                fixUrlNull(posterUrl),
+                null,
+                null,
+                this.name
+            )
+        }
+    }
+
+    override suspend fun load(url: String): LoadResponse? {
+        val response = app.get(url, headers = baseHeaders)
+
+        val document = Jsoup.parse(response.text)
+        val name =
+            document.selectFirst(" h2 ")?.text()?.substringBefore(", page")?.substringBefore("#")
+                ?: return null
+
+        val author =
+            document.selectFirst("#dle-speedbar > div > div > ul > li:nth-child(3) > a > span")
+                ?.text()
+
+        val posterUrl =
+            document.selectFirst("div.box_in > center:nth-child(1) > div > a > img")?.attr("src")
+
+        val data: ArrayList<ChapterData> = ArrayList()
+        val chapters = document.select("div.splitnewsnavigation2.ignore-select > center > div > a")
+
+        data.add(
+            ChapterData(
+                "page 1",
+                url.substringBeforeLast("/") + "/page,1," + url.substringAfterLast("/"),
+                null,
+                null
+            )
+        )
+
+        for (c in 0..(chapters.size / 2)) {
```

Comment:
any reason why /2

Reply:
It only works with it, I tried removing it and figuring out a fix